include/smeagle/parameter.h

```diff
@@ -16,11 +16,28 @@ namespace smeagle {
    * an actual parameter at a callsite, or a return value from a function.
    *
    */
-  struct parameter {
-    std::string name;
-    std::string type;
-    std::string direction;
-    std::string location;
+  class parameter {
+  public:
+    template <typename T> explicit parameter(T x) : self_(std::make_shared<model<T>>(std::move(x))) {}
+
+    std::string facts() const { return self_->facts(); }
```

Comment:
@hainest I think we are probably shooting ourselves in the foot trying to put the core model (what smeagle is about) with some random generation of facts that we might want to use - I think we should nix the facts all together and have Smeagle only focus on the model, and then we can have other tools read in the base models and try different things. To be that's the biggest mistake with libabigail - it doesn't give you an API to interact with the model but goes right to giving you an answer. I think Smeagle should be usable as a library to generate the model facts, but not have some hard coded representation of a random model we happen to be trying. What do you think? It will simplify the stuff here quite a bit - we don't need this complex structure, we can just have a parameter hold the core model facts.